### PR TITLE
Fallback to GET if a server responds with 501 or 503 on HEAD request

### PR DESCRIPTION
**Pull request by meg1502**
Fixes:OX-7063

### CODE REVIEW

File: src/main/java/sirius/biz/storage/layer3/VirtualFile.java [hunk 1]
```diff
@@ -1330,14 +1329,33 @@ public VirtualFile resolveOrLoadChildFromURL(URL url, boolean force, Predicate<S
...
                 path = contentDispositionPath.get();
-            } else {
-                throw Exceptions.createHandled()
-                                .withNLSKey("VirtualFile.loadFromUrl.noValidPath")
-                                .set("url", url)
-                                .handle();
+            } else if (outcall.getResponseCode() == HttpResponseStatus.NOT_IMPLEMENTED.code()
+                       || outcall.getResponseCode() == HttpResponseStatus.SERVICE_UNAVAILABLE.code()) {
+                // some servers will respond with errors on head request, retry with a get request
+                outcall = new Outcall(url);
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer3/VirtualFile.java:1335]: introduce a new variable? or even - extract method?

File: src/main/java/sirius/biz/storage/layer3/VirtualFile.java [hunk 2]
```diff
@@ -1330,14 +1329,33 @@ public VirtualFile resolveOrLoadChildFromURL(URL url, boolean force, Predicate<S
...
+                contentDispositionPath = outcall.parseFileNameFromContentDisposition();
+                lastModifiedHeader = outcall.getHeaderFieldDate(HttpHeaderNames.LAST_MODIFIED.toString());
+                // input has been opened by reading the header and as we don't read the body, close it to free resources
+                outcall.getInput().close();
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer3/VirtualFile.java:1339]: catch an ignore io exceptions here? we don't actually care if close comploetes with or without an exception we still want to continue

**Review by sabieber — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer3/VirtualFile.java [hunk 3]
```diff
@@ -1318,26 +1318,58 @@ public VirtualFile resolveOrLoadChildFromURL(URL url, boolean force, Predicate<S
...
-
-            if (contentDispositionPath.isPresent() && fileExtensionVerifier.test(Files.getFileExtension(
-                    contentDispositionPath.get()))) {
-                path = contentDispositionPath.get();
-            } else {
-                throw Exceptions.createHandled()
-                                .withNLSKey("VirtualFile.loadFromUrl.noValidPath")
-                                .set("url", url)
-                                .handle();
+            Tuple<Optional<String>, Optional<LocalDateTime>> pathAndModifiedHeaders = requestHeaders(url);
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer3/VirtualFile.java:1324]: where is the 2nd part of the tuple used?

Comment (sabieber) [reply src/main/java/sirius/biz/storage/layer3/VirtualFile.java:1324]: In the `doDownload` method below

**Review by sabieber — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**